### PR TITLE
Feature/grpc to rest migration for list metadata apis

### PR DESCRIPTION
**Pull request by Azher2Ali**

### CODE REVIEW

File: src/main/java/org/icgc/argo/program_service/controller/ProgramController.java [hunk 1]
```diff
@@ -137,4 +139,184 @@ public ResponseEntity<List<ProgramsResponseDTO>> listPrograms(
...
+            UUID.fromString(getJoinProgramInviteRequestDTO.getInviteId()));
+    GetJoinProgramInviteResponseDTO getJoinProgramInviteResponseDTO =
+        new GetJoinProgramInviteResponseDTO();
+    getJoinProgramInviteResponseDTO.setInvitation(
+        grpc2JsonConverter.prepareGetJoinProgramInviteResponse(invitation));
+    return new ResponseEntity<GetJoinProgramInviteResponseDTO>(
+        getJoinProgramInviteResponseDTO, HttpStatus.OK);
+  }
+
+  @GetMapping(value = "/listCancers")
```
Comment (joneubank) [inline src/main/java/org/icgc/argo/program_service/controller/ProgramController.java:252]: Can we switch to standard REST syntax? `GET /cancers`

File: src/main/java/org/icgc/argo/program_service/controller/ProgramController.java [hunk 2]
```diff
@@ -137,4 +139,184 @@ public ResponseEntity<List<ProgramsResponseDTO>> listPrograms(
...
+  public ResponseEntity<List<CancerDTO>> listCancers(
+      @Parameter(hidden = true) @RequestHeader(value = "Authorization", required = true)
+          final String authorization) {
+    val listCancersResponse = serviceFacade.listCancers();
+    return new ResponseEntity(
+        grpc2JsonConverter.prepareListCancersResponse(listCancersResponse).getCancers(),
+        HttpStatus.OK);
+  }
+
+  @GetMapping(value = "/listPrimarySites")
```
Comment (joneubank) [inline src/main/java/org/icgc/argo/program_service/controller/ProgramController.java:262]: `GET /primarySites`

File: src/main/java/org/icgc/argo/program_service/controller/ProgramController.java [hunk 3]
```diff
@@ -137,4 +139,184 @@ public ResponseEntity<List<ProgramsResponseDTO>> listPrograms(
...
+          final String authorization) {
+    val listPrimarySitesResponse = serviceFacade.listPrimarySites();
+    return new ResponseEntity(
+        grpc2JsonConverter
+            .prepareListPrimarySitesResponse(listPrimarySitesResponse)
+            .getPrimarySites(),
+        HttpStatus.OK);
+  }
+
+  @GetMapping(value = "/listCountries")
```
Comment (joneubank) [inline src/main/java/org/icgc/argo/program_service/controller/ProgramController.java:274]: `GET /countries`

File: src/main/java/org/icgc/argo/program_service/controller/ProgramController.java [hunk 4]
```diff
@@ -137,4 +139,184 @@ public ResponseEntity<List<ProgramsResponseDTO>> listPrograms(
...
+  public ResponseEntity<List<CountryDTO>> listCountries(
+      @Parameter(hidden = true) @RequestHeader(value = "Authorization", required = true)
+          final String authorization) {
+    val listCountriesResponse = serviceFacade.listCountries();
+    return new ResponseEntity(
+        grpc2JsonConverter.prepareListCountriesResponse(listCountriesResponse).getCountries(),
+        HttpStatus.OK);
+  }
+
+  @GetMapping(value = "/listRegions")
```
Comment (joneubank) [inline src/main/java/org/icgc/argo/program_service/controller/ProgramController.java:284]: `GET /regions`

File: src/main/java/org/icgc/argo/program_service/controller/ProgramController.java [hunk 5]
```diff
@@ -137,4 +139,184 @@ public ResponseEntity<List<ProgramsResponseDTO>> listPrograms(
...
+  public ResponseEntity<List<RegionDTO>> listRegions(
+      @Parameter(hidden = true) @RequestHeader(value = "Authorization", required = true)
+          final String authorization) {
+    val listRegionsResponse = serviceFacade.listRegions();
+    return new ResponseEntity(
+        grpc2JsonConverter.prepareListRegionsResponse(listRegionsResponse).getRegions(),
+        HttpStatus.OK);
+  }
+
+  @GetMapping(value = "/listInstitutions")
```
Comment (joneubank) [inline src/main/java/org/icgc/argo/program_service/controller/ProgramController.java:294]: `GET /institutions`

File: src/main/java/org/icgc/argo/program_service/controller/ProgramController.java [hunk 6]
```diff
@@ -137,4 +139,184 @@ public ResponseEntity<List<ProgramsResponseDTO>> listPrograms(
...
+          final String authorization) {
+    val listInstitutionsResponse = serviceFacade.listInstitutions();
+    return new ResponseEntity(
+        grpc2JsonConverter
+            .prepareListInstitutionsResponse(listInstitutionsResponse)
+            .getInstitutions(),
+        HttpStatus.OK);
+  }
+
+  @PostMapping(value = "/addInstitutions")
```
Comment (joneubank) [inline src/main/java/org/icgc/argo/program_service/controller/ProgramController.java:306]: `POST /institutions`

**Review by joneubank — CHANGES_REQUESTED** [review]
Requesting simplification of the API paths

File: src/main/java/org/icgc/argo/program_service/controller/ProgramController.java (hunk 1 above)
Comment (Azher2Ali) [reply src/main/java/org/icgc/argo/program_service/controller/ProgramController.java]: Updated as per review comment

File: src/main/java/org/icgc/argo/program_service/controller/ProgramController.java (hunk 2 above)
Comment (Azher2Ali) [reply src/main/java/org/icgc/argo/program_service/controller/ProgramController.java]: Updated as per review comment

File: src/main/java/org/icgc/argo/program_service/controller/ProgramController.java (hunk 3 above)
Comment (Azher2Ali) [reply src/main/java/org/icgc/argo/program_service/controller/ProgramController.java]: Updated as per review comment

File: src/main/java/org/icgc/argo/program_service/controller/ProgramController.java (hunk 4 above)
Comment (Azher2Ali) [reply src/main/java/org/icgc/argo/program_service/controller/ProgramController.java]: Updated as per review comment

File: src/main/java/org/icgc/argo/program_service/controller/ProgramController.java (hunk 5 above)
Comment (Azher2Ali) [reply src/main/java/org/icgc/argo/program_service/controller/ProgramController.java]: Updated as per review comment

File: src/main/java/org/icgc/argo/program_service/controller/ProgramController.java (hunk 6 above)
Comment (Azher2Ali) [reply src/main/java/org/icgc/argo/program_service/controller/ProgramController.java]: Updated as per review comment

**Review by joneubank — APPROVED** [review]

**Pull request merged**